### PR TITLE
Ported component AI to the new component system

### PR DESCRIPTION
**Pull request by DamianX**
It's still almost completely broken, but slightly less so. The NPC human can now _sometimes_ walk toward its intended target, and correctly equip the items it needs once again. But that's about it. Everything else is still broken.

### CODE REVIEW

File: code/_hooks/events.dm [hunk 1]
```diff
@@ -266,12 +288,11 @@
 	if(!length(event_handlers))
 		// This datum does not have any handler registered for this event_type.
 		return
-	. = NONE
 	for(var/key in event_handlers)
 		var/list/handler = event_handlers[key]
 		var/objRef = handler[EVENT_HANDLER_OBJREF_INDEX]
 		var/procName = handler[EVENT_HANDLER_PROCNAME_INDEX]
-		. |= CallAsync(objRef, procName, arguments)
```
Comment (Exxion) [inline code/_hooks/events.dm]: What's the difference here?

Comment (Exxion) [reply code/_hooks/events.dm]: Pretend this is on the new line rather than the old one. For some reason the comment button wouldn't show up on the new one

Comment (DamianX) [reply code/_hooks/events.dm]: `null |= list()` is a runtime error

`null = null | list()` is fine

Comment (Exxion) [reply code/_hooks/events.dm]: Ah, right. Might want to add a comment explaining that because it looks pretty arcane

**Comment by Kurfursten** [comment]
confl

**Pull request merged**